src/containerapp/azext_containerapp/_utils.py

```diff
@@ -48,10 +55,15 @@ def register_provider_if_needed(cmd, rp_name):
         _register_resource_provider(cmd, rp_name)
 
 
-def validate_container_app_name(name):
-    if name and len(name) > MAXIMUM_CONTAINER_APP_NAME_LENGTH:
-        raise ValidationError(f"Container App names cannot be longer than {MAXIMUM_CONTAINER_APP_NAME_LENGTH}. "
-                              f"Please shorten {name}")
+def validate_container_app_name(name, appType):
+    if (appType == AppType.ContainerAppJob.name):
+        if name and len(name) > MAXIMUM_CONTAINER_APP_NAME_LENGTH:
+            raise ValidationError(f"Container App Job names cannot be longer than {MAXIMUM_CONTAINER_APP_NAME_LENGTH}. "
+                                  f"Please shorten {name}")
+    if (appType == AppType.ContainerApp.name):
+        if name and len(name) > MAXIMUM_CONTAINER_APP_NAME_LENGTH:
+            raise ValidationError(f"Container App Job names cannot be longer than {MAXIMUM_CONTAINER_APP_NAME_LENGTH}. "
+                                  f"Please shorten {name}")
```

Comment:
why we use one validator for both container app and container app job? Is it possible to separate them

Reply:
@Juliehzl basically the validation on the name in the RP is also the same for Apps and Jobs and hence I have combined here as well.